### PR TITLE
WIP feat: add webkms client support in web-wallet

### LINKED ISSUE #541: integrate webkms client into edge-agent

### PR DESCRIPTION
**Pull request by Baha-sk**
closes #541

Signed-off-by: Baha Shaaban <baha.shaaban@securekey.com>

### CODE REVIEW

File: cmd/wallet-server/startcmd/start.go [hunk 1]
```diff
@@ -93,6 +93,10 @@ const (
 	keyEDVURLFlagName  = "key-edv-url"
 	keyEDVURLFlagUsage = "Operational key EDV Server URL"
 	keyEDVURLEnvKey    = "HTTP_SERVER_KEY_EDV_URL"
+
+	useRemoteKMSFlagName  = "use-remote-kms"
+	useRemoteKMSFlagUsage = "Use Remote KMS? (y/n)"
+	useRemoteKMSEnvKey    = "USE_REMOTE_KMS"
```
Comment (llorllale) [inline cmd/wallet-server/startcmd/start.go:99]: Followup: just like we have the concept of "database type" in the configuration, we should have the same for KMS. Its value should be "webkms" in this case.

File: cmd/wallet-server/startcmd/start.go [hunk 2]
```diff
@@ -522,10 +528,24 @@ func getKeyServerParams(cmd *cobra.Command) (*keyServerParameters, error) {
...
+	var useRemoteKMS bool
+
+	useRemoteKMSStr, err := cmdutils.GetUserSetVarFromString(
+		cmd, useRemoteKMSFlagName, useRemoteKMSEnvKey, true)
+	if err != nil {
+		return nil, fmt.Errorf("use remote kms : %w", err)
+	}
+
+	switch useRemoteKMSStr {
+	case "Y", "y", "yes", "YES", "true", "TRUE":
```
Comment (llorllale) [inline cmd/wallet-server/startcmd/start.go:540]: Simplify this to just `true` or `false` and use `strconv.ParseBool()`

File: pkg/restapi/oidc/operations.go [hunk 3]
```diff
@@ -599,12 +603,31 @@ func (o *Operation) onboardUser(sub, accessToken string) (string, error) { // no
 		}
 	}
 
+	edvOpsKID, err := createKey(o.keyServer.OpsKMSURL, getKeystoreID(opsKeyStoreURL), kms.ECDH256KWAES256GCM, h,
```
Comment (llorllale) [inline pkg/restapi/oidc/operations.go:606]: what is the use of this key?

**Review by llorllale — APPROVED** [review]

File: cmd/wallet-server/startcmd/start.go (hunk 1 above)
Comment (Baha-sk) [reply cmd/wallet-server/startcmd/start.go:99]: it's just a boolean for now, either local or remote, that's a possible idea though, if we want to match the client kms implementation with the server's implementation. Maybe it's good for a future release.

File: cmd/wallet-server/startcmd/start.go (hunk 2 above)
Comment (Baha-sk) [reply cmd/wallet-server/startcmd/start.go]: updated

File: cmd/wallet-server/startcmd/start.go (hunk 1 above)
Comment (Baha-sk) [reply cmd/wallet-server/startcmd/start.go:99]: it has been update to kmsType (in a separate PR)

**Comment by Baha-sk** [comment]
closing this PR as it's now stale

**Pull request closed without merging**